orc8r/cloud/go/services/analytics/collector_servicer.go

```diff
@@ -18,32 +18,127 @@ import (
 	"magma/orc8r/cloud/go/services/analytics/calculations"
 	"magma/orc8r/cloud/go/services/analytics/protos"
 	"magma/orc8r/cloud/go/services/analytics/query_api"
+	"magma/orc8r/lib/go/metrics"
 
 	"github.com/golang/glog"
 )
 
 //CollectorService implements the operations of collecting the metrics from CWF service
 type CollectorService struct {
-	calculations  []calculations.Calculation
-	promAPIClient query_api.PrometheusAPI
+	calculations    []calculations.Calculation
+	promAPIClient   query_api.PrometheusAPI
+	userStateMgr    *calculations.UserStateMgr
+	analyticsConfig *calculations.AnalyticsConfig
 }
 
 //NewCollectorService constructs new collector service
-func NewCollectorService(promAPIClient query_api.PrometheusAPI, calculations []calculations.Calculation) *CollectorService {
-	return &CollectorService{promAPIClient: promAPIClient, calculations: calculations}
+func NewCollectorService(analyticsConfig *calculations.AnalyticsConfig, promAPIClient query_api.PrometheusAPI, calculations []calculations.Calculation,
+	userStateMgr *calculations.UserStateMgr) *CollectorService {
+	return &CollectorService{promAPIClient: promAPIClient, calculations: calculations, analyticsConfig: analyticsConfig, userStateMgr: userStateMgr}
+}
+
+// filterResults filters the results to ensure that the user aggregate metrics get filtered out in
+// case the number of users in that context falls below minimum threshold
+func (svc *CollectorService) filterResults(results []*protos.CalculationResult) []*protos.CalculationResult {
+	userStateMgr := svc.userStateMgr
+	// nothing to filter
+	if userStateMgr == nil {
+		return results
+	}
+
+	filteredResults := []*protos.CalculationResult{}
+	analyticsConfig := svc.analyticsConfig
+	for _, result := range results {
```

Comment:
This is waaaay too much nesting. Find a way to clean this up so us mere mortals can read through it

Reply:
Will clean this up.